### PR TITLE
added necessary fields for event pulling and an event example

### PR DESCRIPTION
**Pull request by ozdoganoguzhan**
**Type Changes:** Added some fields that are required to create a subscription for pulling messages with a filter, and also for pulling messages from that subscription. 

**New Example:** Added an example on how to pull messages periodically from a specific camera. Tested on Dahua.

Screenshot_10

Solves #107

### CODE REVIEW

File: onvif/Cargo.toml [hunk 1]
```diff
@@ -36,6 +36,7 @@ xsd-macro-utils = { git = "https://github.com/lumeohq/xsd-parser-rs", rev = "7f3
 xsd-types = { git = "https://github.com/lumeohq/xsd-parser-rs", rev = "7f3d433" }
 yaserde = "0.7.1"
 yaserde_derive = "0.7.1"
+b_2 = {path = "../wsdl_rs/b_2"}
```
Comment (DmitrySamoylov) [inline onvif/Cargo.toml]: It's only used in the example, so let's move it to `[dev-dependencies]`.

File: wsdl_rs/b_2/src/lib.rs [hunk 2]
```diff
@@ -131,15 +137,45 @@ pub struct NotificationMessageHolderType {
 
 impl Validate for NotificationMessageHolderType {}
 
+#[derive(Default, PartialEq, Debug, YaSerialize, YaDeserialize)]
+#[yaserde(prefix = "tt", namespace = "tt: http://www.onvif.org/ver10/schema")]
+
```
Comment (DmitrySamoylov) [inline wsdl_rs/b_2/src/lib.rs]: Unneeded newline.

File: wsdl_rs/b_2/src/lib.rs [hunk 3]
```diff
@@ -36,7 +39,10 @@ impl Validate for TopicExpressionType {}
     prefix = "wsnt",
     namespace = "wsnt: http://docs.oasis-open.org/wsn/b-2"
 )]
-pub struct FilterType {}
+pub struct FilterType {
+    #[yaserde(prefix = "wsnt", rename = "TopicExpression")]
+    pub topic_expression: TopicExpressionType,
```
Comment (DmitrySamoylov) [inline wsdl_rs/b_2/src/lib.rs]: In WSDL, it's a sequence of `any` types:

```
  <xsd:complexType name="FilterType">
    <xsd:sequence>
      <xsd:any minOccurs="0" maxOccurs="unbounded"/>
    </xsd:sequence>
  </xsd:complexType>
```

so it shouldn't be a required field. For a lack of better handling of `any` types, let's make the field optional to not break other use-cases:

```
pub topic_expression: Option<TopicExpressionType>,
```

File: wsdl_rs/b_2/src/lib.rs [hunk 4]
```diff
@@ -131,15 +137,45 @@ pub struct NotificationMessageHolderType {
...
+        pub simple_item: SimpleItemType,
+    }
+
+    #[derive(Default, PartialEq, Debug, YaSerialize, YaDeserialize)]
+    #[yaserde(prefix = "tt", namespace = "tt: http://www.onvif.org/ver10/schema")]
+    pub struct MessageTypeInner {
+        #[yaserde(prefix = "tt", rename = "Data")]
+        pub data: DataType,
+    }
+
```
Comment (DmitrySamoylov) [inline wsdl_rs/b_2/src/lib.rs]: Do you have an example XML from the camera? If yes, can you also add a test [here](https://github.com/lumeohq/onvif-rs/blob/a49b43d4713ee5ff5ffb7bab9105d6793c6c1085/schema/src/tests.rs)?

**Review by DmitrySamoylov — CHANGES_REQUESTED** [review]

File: wsdl_rs/b_2/src/lib.rs (hunk 2 above)
Comment (ozdoganoguzhan) [reply wsdl_rs/b_2/src/lib.rs]: removed

File: onvif/Cargo.toml (hunk 1 above)
Comment (ozdoganoguzhan) [reply onvif/Cargo.toml]: done

File: wsdl_rs/b_2/src/lib.rs (hunk 3 above)
Comment (ozdoganoguzhan) [reply wsdl_rs/b_2/src/lib.rs]: done

File: wsdl_rs/b_2/src/lib.rs (hunk 4 above)
Comment (ozdoganoguzhan) [reply wsdl_rs/b_2/src/lib.rs]: Added 2 tests: one for pulling messages one for creating subscription and getting the address from that subscription.

Thank you for great feedback!

**Comment by DmitrySamoylov** [comment]
Can you please rebase on `main` to see if CI is green?

**Comment by asuper0** [comment]
hi, how is it going?

**Comment by ozdoganoguzhan** [comment]
The problem was in the test XML. I removed the envelope declarations and just left the responses and put the namespaces in them. Also moved the related tests to the end of tests file, to prevent conflict with the more recent commits to test file.

**Comment by DmitrySamoylov** [comment]
> error: useless use of `vec!`

This was fixed in `main`. Please, rebase again.

**Comment by sqrtsanta** [comment]
Sorry for chipping in, also very interested in subscriptions & pull_messages. Really appreciate your work folks!

**Comment by DmitrySamoylov** [comment]
Not sure if you receive notifications on failing CI jobs, but clippy is not happy now with some of the changes.

**Comment by ozdoganoguzhan** [comment]
everything should be ok now, thanks for your patience 😄

**Review by DmitrySamoylov — APPROVED** [review]

**Pull request merged**